index.d.ts

```diff
@@ -642,7 +642,17 @@ declare namespace plugins {
    * This plugin automatically instruments the
    * [elasticsearch](https://github.com/elastic/elasticsearch-js) module.
    */
-  interface elasticsearch extends Instrumentation {}
+  interface elasticsearch extends Instrumentation {
+    /**
+     * Hooks to run before spans are finished.
+     */
+    hooks?: {
+      /**
+       * Hook to execute just before the query span finishes.
+       */
+      query?: (span?: opentracing.Span, params?: any) => any;
```

Comment:
The type of params should probably be `TransportRequestParams` from [Transport.d.ts](https://github.com/elastic/elasticsearch-js/blob/master/lib/Transport.d.ts).

Reply:
I pulled `@elastic/elasticsearch` in as a dev dependency so I could import it in this test. Though let me know if you prefer me to simply recreate the interface rather than add another package into the mix.

Reply:
Ideally it would be best to use the types from the library in case they change over time. While it would be better to avoid having to add the library as a dev dependency, this is a limitation that will be addressed later for all plugins.

Reply:
Ah okay, that's fine. How should I get around the licensing restriction in the linter?